tests/bigquery/pipeline_tests.py

```diff
@@ -0,0 +1,228 @@
+    self.assertEqual(actual_load_config['mode'], 'append')
+
+    input_config = {
+      'table': 'test_table',
+      'schema': 'test_schema'
+    }
+    actual_load_config = bq._get_load_parameters(input_config)
+    self.assertIsNone(actual_load_config)
+
+    input_config = {
+      'schema': 'test_schema'
+    }
+    actual_load_config = bq._get_load_parameters(input_config)
+    self.assertIsNone(actual_load_config)
+
+    input_config = {
+      'path': 'test_path',
+    }
+    actual_load_config = bq._get_load_parameters(input_config)
+    expected_load_config = {
+      'type': 'pydatalab.bq.load',
+      'format': 'csv',
+      'csv_options': None,
+      'path': 'test_path',
+    }
+    self.assertDictEqual(actual_load_config, expected_load_config)
+
+    input_config = {
+      'path': 'test_path',
+      'format': 'json'
+    }
+    actual_load_config = bq._get_load_parameters(input_config)
+    expected_load_config = {
+      'type': 'pydatalab.bq.load',
+      'format': 'json',
+      'path': 'test_path',
+      'csv_options': None
+    }
+    self.assertDictEqual(actual_load_config, expected_load_config)
+
+  def test_get_extract_parameters(self):
+    input_config = {
+      'path': 'test_path',
+      'table': 'test_table',
+    }
+    execute_config = {}
+    actual_extract_config = bq._get_extract_parameters('foo_execute_task', execute_config,
+                                                       input_config)
+    expected_extract_config = {
+      'type': 'pydatalab.bq.extract',
+      'up_stream': ['foo_execute_task'],
+      'format': 'csv',
+      'csv_options': None,
+      'path': 'test_path',
+      'table': 'test_table',
+    }
+
+    self.assertDictEqual(actual_extract_config, expected_extract_config)
+    self.assertDictEqual(execute_config, {'table': 'test_table'})
+
+  @mock.patch('google.datalab.Context.default')
+  @mock.patch('google.datalab.utils.commands.notebook_environment')
+  @mock.patch('google.datalab.bigquery.Table.exists')
+  @mock.patch('google.datalab.bigquery.commands._bigquery._get_table')
+  def test_pipeline_cell_golden(self, mock_get_table, mock_table_exists, mock_environment,
+                                mock_default_context):
+    table = google.datalab.bigquery.Table('project.test.table')
+    mock_get_table.return_value = table
+    mock_table_exists.return_value = True
+    context = TestCases._create_context()
+    mock_default_context.return_value = context
+
+    env = {
+      'foo_query': google.datalab.bigquery.Query(
+        'SELECT * FROM publicdata.samples.wikipedia LIMIT 5')
+    }
+    mock_environment.return_value = env
+
+    args = {'name': 'bq_pipeline_test', 'debug': True}
+    # TODO(rajivpb): The references to foo_query need to be resolved.
+    cell_body = """
+            schedule:
+                start: 2009-05-05T22:28:15Z
+                end: 2009-05-06T22:28:15Z
+                interval: '@hourly'
+            input:
```

Comment:
Looking at this in its entirety:

1. I think csv can be simplified (header-lines, delimiter)
2. Seeing path and table together is confusing. I think the scenarios we are about are:

a) path only (external datasource)
b) table only
c) path -> load table -> use table (add a new field, `import` instead of table)

Reply:
https://github.com/googledatalab/pydatalab/issues/544